handlers.go

```diff
@@ -711,39 +707,11 @@ func (s *Server) handleAddIpfs(c echo.Context, u *User) error {
 func (s *Server) handleAddCar(c echo.Context, u *User) error {
 	ctx := c.Request().Context()
 
-	if s.CM.contentAddingDisabled || u.StorageDisabled || s.CM.localContentAddingDisabled {
-		if s.CM.localContentAddingDisabled {
-			uep, err := s.getPreferredUploadEndpoints(u)
-			if err != nil {
-				log.Warnf("failed to get preferred upload endpoints: %s", err)
-			} else if len(uep) > 0 {
-				// propagate any query params
-				req, err := http.NewRequest("POST", fmt.Sprintf("%s/content/add-car", uep[rand.Intn(len(uep))]), c.Request().Body)
-				if err != nil {
-					return err
-				}
-				req.Header = c.Request().Header.Clone()
-				req.URL.RawQuery = c.Request().URL.Query().Encode()
-
-				resp, err := http.DefaultClient.Do(req)
-				if err != nil {
-					return err
-				}
-
-				c.Response().WriteHeader(resp.StatusCode)
-
-				_, err = io.Copy(c.Response().Writer, resp.Body)
-				if err != nil {
-					log.Errorf("proxying content-add body errored: %s", err)
-				}
-			} else {
-				return &util.HttpError{
-					Code:    http.StatusBadRequest,
-					Reason:  util.ERR_CONTENT_ADDING_DISABLED,
-					Details: "uploading content to this node is not allowed at the moment",
-				}
-			}
-		}
+	if err := s.redirectIfLocalContentAddingDisabled(c, u); err != nil {
```

Comment:
you should call `ErrorIfContentAddingDisabled` before `redirectIfLocalContentAddingDisabled`

Reply:
also for `redirectIfLocalContentAddingDisabled`, if true, code should redirect and not proceed any further

Reply:
yes, it is a different behavior but that should have been the proper order.